pkgs/misc/uboot/default.nix

```diff
@@ -607,6 +607,14 @@ in {
     filesToInstall = ["u-boot-sunxi-with-spl.bin"];
   };
 
+  ubootTuringRK1RK3588 = buildUBoot {
+    defconfig = "turing-rk1-rk3588_defconfig";
+    extraMeta.platforms = [ "aarch64-linux" ];
+    BL31 = "${armTrustedFirmwareRK3588}/bl31.elf";
+    ROCKCHIP_TPL = rkbin.TPL_RK3588;
+    filesToInstall = [ "u-boot.itb" "idbloader.img" "u-boot-rockchip.bin" ];
```

Comment:
This shows that a SPI flash binary should also be compiled for the Turing Pi RK1. Please also add `"u-boot-rockchip-spi.bin"` ot the `filesToInstall` list. :)

https://github.com/u-boot/u-boot/blob/25049ad560826f7dc1c4740883b0016014a59789/configs/turing-rk1-rk3588_defconfig#L17

Reply:
I don't think it has an SPI flash. None of the technical spec listing, the u-boot commit introducing it (153ac950a59960967c752177ceda9fec1c55fc98), or any of the kernel refers to it in any way.

It still being enabled in the defconfig might just be a copypaste from another defconfig. I can ask back, but don't think we should add this on suspicion in the meantime.

Reply:
I'm not a DT expert, maybe @samueldr can weigh in here but I do see a node for SPI on the RK1: https://github.com/u-boot/u-boot/blob/25049ad560826f7dc1c4740883b0016014a59789/arch/arm/dts/rk3588-turing-rk1.dtsi#L283

Almost similar to the Rkcok 5B's DT: https://github.com/u-boot/u-boot/blob/25049ad560826f7dc1c4740883b0016014a59789/arch/arm/dts/rk3588-rock-5b.dts#L409

Reply:
First of all, SPI != SPI Flash module. `&spi2` here is used to communicate with the PMIC.

Neither U-Boot nor Linux (from which U-Boot syncs regularly) declare an SPI Flash module.

Though, with that said, there are no "schematics" I can find, no list of hardware. Even the specs list is uh... incomplete.

The photos on the site are terribly crushed. @flokli: what's the chip at U18?

Though, maybe we'll have more info soon, *someone* asked for more info about that topic.

 - https://lore.kernel.org/all/4lspwhoesvjvbaxlsx4fl6fbsnfte2qbpq2tmfupdpw5jh6vqp@oculnryb7ue6/

Reply:
It's already nicely packaged inside a case, and from outside I can only get you more blurry pictures.

In any case, let's not block the PR over this (and push it to another staging round). We can update the leaf turing expression in case we find an SPI flash somewhere on the board ;-)

Reply:
Upstream replied, there is no SPI flash soldered on, only a unpopulated pad. They agree the `CONFIG_ROCKCHIP_SPI_IMAGE=y` might not be a sensible default.